catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/DeleteOperations.java

```diff
@@ -0,0 +1,408 @@
+            LOGGER.warn(PRE_INGEST_ERROR + see.getMessage(), see);
+            throw new IngestException(PRE_INGEST_ERROR + see.getMessage());
+
+        } catch (RuntimeException re) {
+            LOGGER.warn("Exception during runtime while performing delete", re);
+            throw new InternalIngestException("Exception during runtime while performing delete");
+
+        } finally {
+            if (deleteStorageRequest != null) {
+                try {
+                    storageSupplier.get()
+                            .commit(deleteStorageRequest);
+                } catch (StorageException e) {
+                    LOGGER.error("Unable to remove stored content items.", e);
+                }
+            }
+        }
+
+        return deleteResponse;
+    }
+
+    //
+    // Private helper methods
+    //
+
+    private DeleteResponse injectAttributes(DeleteResponse response) {
+        List<Metacard> deletedMetacards = response.getDeletedMetacards()
+                .stream()
+                .map((original) -> opsMetacardSupport.applyInjectors(original,
+                        frameworkProperties.getAttributeInjectors()))
+                .collect(Collectors.toList());
+
+        return new DeleteResponseImpl(response.getRequest(),
+                response.getProperties(),
+                deletedMetacards,
+                response.getProcessingErrors());
+    }
+
+    /**
+     * Validates that the {@link DeleteRequest} is non-null, has a non-empty list of
+     * {@link Metacard}s in it, and a non-null attribute name (which specifies if the delete is
+     * being done by product URI or ID).
+     *
+     * @param deleteRequest the {@link DeleteRequest}
+     * @throws IngestException if the {@link DeleteRequest} is null, or has null or empty {@link Metacard} list,
+     *                         or a null attribute name
+     */
+    private void validateDeleteRequest(DeleteRequest deleteRequest) throws IngestException {
+        if (deleteRequest == null) {
+            throw new IngestException(
+                    "DeleteRequest was null, either passed in from endpoint, or as output from PreIngestPlugins");
+        }
+        List<?> entries = deleteRequest.getAttributeValues();
+        if (CollectionUtils.isEmpty(entries) || deleteRequest.getAttributeName() == null) {
+            throw new IngestException(
+                    "Cannot perform delete with null/empty attribute value list or null attributeName, "
+                            + "either passed in from endpoint, or as output from PreIngestPlugins");
+        }
+    }
+
+    private DeleteResponse doRemoteDelete(DeleteRequest deleteRequest) {
+        HashSet<ProcessingDetails> exceptions = new HashSet<>();
+        Map<String, Serializable> properties = new HashMap<>();
+
+        List<CatalogStore> stores = opsCrudSupport.getCatalogStoresForRequest(deleteRequest,
+                exceptions);
+
+        List<Metacard> metacards = new ArrayList<>();
```

Comment:
I'm just very tentative about changing collections that get passed around that much into immutable versions without a full analysis. We have getters returning the collections, not copies of them, in several places. This collection gets passed into a ctor for one object, it has a getter for the collection which gets passed into _other_ ctors in other places. I like the idea of swapping it out but in practice it concerns me.